src/main/java/service/PrintWeather.java

```diff
@@ -1,35 +1,46 @@
 package service;
 
-import lombok.Data;
-import model.FormatDate;
-import model.List;
-import model.ResponseModel;
+import model.ReplyJsonList;
+import model.Response;
 
-@Data
 public class PrintWeather {
 
-    private ResponseModel responseModel;
+    private final Response response;
+    private final OptionCli optionCli;
 
-    public PrintWeather(ResponseModel responseModel) {
-        this.responseModel = responseModel;
+    public PrintWeather(OptionCli optionCli, Response response) {
+        this.optionCli = optionCli;
+        this.response = response;
     }
 
-    public void printCustomWeather() {
-        System.out.println(this.responseModel);
+    private void printCurrentWeather() {
+        System.out.println(this.response);
     }
 
-    public void searchAndPrintWeatherForDataTime(String dataTime) {
-        for (List search : this.responseModel.getList()) {
-            if (FormatDate.unixToLocalDateString(search.getDt()).equals(dataTime)) {
-                System.out.println(search.toString() + this.responseModel.getCity());
+    private void printWeatherForDateTime(String dateTime) {
+        for (ReplyJsonList search : this.response.getReplyJsonList()) {
+            if (FormatDate.unixToLocalDateAndTimeString(search.getDt()).equals(dateTime)) {
+                System.out.println(search.toString() + this.response.getCity());
             }
         }
     }
 
-    public void printAvailableForecastDataTimes() {
-        for (List search : this.responseModel.getList()) {
-            System.out.println(FormatDate.unixToLocalDateString(search.getDt()));
+    private void printAvailableForecastDateTimes() {
+        for (ReplyJsonList search : this.response.getReplyJsonList()) {
+            System.out.println(FormatDate.unixToLocalDateAndTimeString(search.getDt()));
         }
     }
 
+    public void print() {
+
+        if (optionCli.getDataTime() != null) {
+            printWeatherForDateTime(optionCli.getDataTime());
+        } else if (response.getReplyJsonList() != null) {
+            printAvailableForecastDateTimes();
+        } else {
+            printCurrentWeather();
+        }
+
+    }
```

Comment:
to jednak bym zostawił tam gdzie było pownieważ znów nie widze logicznego powiązania optionCli z printWriterem więc nie ma sensu go oczekiwac jako parametr w konstrutorze